Added 'disabled' styling to switches.

jellisii opened the pull request:
I've run up on some situations where I need to fix a switch into a position.  This adds CSS to reflect the disabled attribute on the input.

jellisii committed:
Added 'disabled' styling to switches.

DanielRuf commented:
id, for and name should be different for the second example.

Please use the master branch for changes in the docs.

jellisii committed:
Updated name of new switches.

ncoden commented:
@DanielRuf We do not want to change the docs on `master` as it describe a new feature on `develop`.

ncoden commented on scss/components/_switch.scss:
```diff
@@ -208,6 +208,10 @@ $switch-paddle-transition: all 0.25s ease-out !default;
   .switch-input {
     @include switch-input;
+      // indicate a disabled switch
+    &:disabled ~ label {
```
I don't like this selector (non BEM-like) but this is the CSS architecture of the component, so good enough.

ncoden commented on scss/components/_switch.scss:
```diff
@@ -208,6 +208,10 @@ $switch-paddle-transition: all 0.25s ease-out !default;
   // <input> element
   .switch-input {
     @include switch-input;
+      // indicate a disabled switch
```
This is not aligned ;)

ncoden commented on scss/components/_switch.scss:
```diff
@@ -208,6 +208,10 @@ $switch-paddle-transition: all 0.25s ease-out !default;
     @include switch-input;
+      // indicate a disabled switch
+    &:disabled ~ label {
+      cursor: not-allowed;
```
You can use `$switch-cursor-disabled` instead of a hard-coded property (many people don't like the `not-allowed` cursor.

Also, you could add `$switch-opacity-disabled` (with `.5` or `.66`) as default so we can recognize a disabled switch without having to move the cursor on it.

DanielRuf commented:
> @DanielRuf We do not want to change the docs on master as it describes a new feature on develop.

Ah ok =) So when it comes with new changes, it should be in the same branch (develop).

ncoden commented:
If we want to improve the doc of the last version, we push on master so it can be deployed quickly. If we improve/add some docs for features/components of the next version, we keep it on develop and it will made be public when the next version is released.

jellisii committed:
Changes requested by @ncoden

Added variables for switch-cursor-disabled and switch-opacity-disabled.

Altered css selector to use class instead of element.

Added reduced opacity for disabled switches.

Fixed line spacing for comment.

ncoden commented on scss/components/_switch.scss:
```diff
@@ -208,6 +216,11 @@ $switch-paddle-transition: all 0.25s ease-out !default;
+    &:disabled ~ .switch-paddle {
+      cursor: $switch-cursor-disabled;
+      opacity: $switch-opacity-disabled;
+    }
```
Sorry for not putting that in my first review, but could you move that to `@mixin switch-paddle`, just after the `input:checked` styles ?

ncoden reviewed (commented):
Thanks for the changes :)

jellisii committed:
Moved disabled settings into switch-paddle mixin.

ncoden reviewed (approved):
👍  Thanks @jellisii. I'll resolve conflicts.

ncoden committed:
Merge branch 'develop' into develop

jellisii commented:
Thank you @ncoden and @DanielRuf .

ncoden commented:
@jellisii  For your next pull request, you can push your work to a dedicated branch (like `feat/add-switch-disabled-state` and create a pull request from it. So you can continue working on your develop branch without polluting your existing pull requests.

ncoden commented:
Preview:
![screen shot 2018-02-26 at 23 44 11](https://user-images.githubusercontent.com/9939075/36700086-fcd1b550-1b4e-11e8-8a8f-eb609050065b.png)

ncoden commented:
Everything looks good so I'm merging this.
Thanks again @jellisii 👍 .
